### DISCUSSION TITLE
Need help configuring Zigbee2MQTT/configuration.yaml

### DISCUSSION THREAD

**Discussion by Ursusprimus** [Koenkk/zigbee2mqtt#7179] [Devices]
Hello,

I tried getting Zigbee2MQTT to run on my RasPi3 using a ZZH but (being unfamiliar with Linux and Network settings) without success. 
The problem seems to be the server-address. I tried: server: 'mqtt://localhost' and 192.168.178.56 and  '192.168.178.56' and 'mqtt://192.168.178.56:1883' and   mqtt://192.168.178.56:1883 and  mqtt://192.168.178.56 and  'mqtt://192.168.178.56'  All of this leads to this feedback:

`pi@raspberrypi:/opt/zigbee2mqtt $ npm start

> zigbee2mqtt@1.18.2 start /opt/zigbee2mqtt
> node index.js

(node:18781) UnhandledPromiseRejectionWarning: YAMLException: bad indentation of a mapping entry at line 11, column 2:
     server: 'mqtt://192.168.178.56'
     ^
    at generateError (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:167:10)
    at throwError (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:173:9)
    at readBlockMapping (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:1107:7)
    at composeNode (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:1359:12)
    at readDocument (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:1525:3)
    at loadDocuments (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:1588:5)
    at load (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:1614:19)
    at Object.safeLoad (/opt/zigbee2mqtt/node_modules/js-yaml/lib/js-yaml/loader.js:1637:10)
    at Object.read (/opt/zigbee2mqtt/lib/util/yaml.js:7:21)
    at read (/opt/zigbee2mqtt/lib/util/settings.js:274:20)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18781) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:18781) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

And here's my configuration:  (in this post (not in the file) I replaced "#" with "-" because of formatting)

- Required: Home Assistant integration (MQTT discovery)
homeassistant: false
- Required: allow new devices to join.
- WARNING: Disable this after all devices have been paired!
permit_join: true
- Required: MQTT settings
mqtt:
 - Required: MQTT base topic for zigbee2mqtt MQTT messages
base_topic: zigbee2mqtt
 - Required: MQTT server URL  - 
 server: 192.168.178.56
 - Optional: MQTT server authentication user - 
 user: Ursusprimus
 - Optional: MQTT server authentication password  - 
 password: Ursusprimus
 - Optional: MQTT client ID
 client_id: 'Zigbee'
 - Disable self-signed SSL certificates
 reject_unauthorized: true
 - Optional: Include device information to mqtt messages (default: false)  include_device_information: true
- Required: serial settings
serial:
 - Required: location of CC2531 USB sniffer  port:  
/dev/ttyUSB0
 - Optional: disable LED of CC2531 USB sniffer  disable_led: false

Since I'm totally stuck, I'd greatly appreciate help :-)

**Comment by Twit123** [comment]
Hi,

You need to post your yaml using the <> code button, then we can see what your config actually looks like,.

Looking at the error, I think you have a spacing issue.

Mine looks like this (just the necessary parts)

```
data_path: /share/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://core-mosquitto'
  user: hass
  password: Russel12
serial:
  port: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
```

Notice the server, base topic, user and password are indented which makes them part of the "mqtt:" section

Since I run docker I use "server: 'mqtt://core-mosquitto'" but in your case the IP should work or 127.0.0.1.

Simon

**Reply by Ursusprimus** [reply]
Hello Simon,

thanks for trying to help! When it comes to formatting: I can't seem to get line breaks. Not even with two spaces and manual enter at the end of each line as described in some postings. So I'll add screenshots just to make sure.
I tried some changes and paid attention to the spacing, but without success.

My current configuration: 
![image](https://user-images.githubusercontent.com/79019508/116261640-2266ab00-a778-11eb-8aba-e488ef3a3c03.png)

Which leads to this error message:
![image](https://user-images.githubusercontent.com/79019508/116262358-c2243900-a778-11eb-832a-4fd8dfb1c215.png)

And the according logfile (some lines are strangely formatted as crossed out, just ignore it, in the original they're normal):
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
2 info using npm@6.14.12
3 info using node@v14.16.1
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle zigbee2mqtt@1.18.2~prestart: zigbee2mqtt@1.18.2
6 info lifecycle zigbee2mqtt@1.18.2~start: zigbee2mqtt@1.18.2
7 verbose lifecycle zigbee2mqtt@1.18.2~start: unsafe-perm in lifecycle true
8 verbose lifecycle zigbee2mqtt@1.18.2~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/opt/zigbee2mqtt/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games
9 verbose lifecycle zigbee2mqtt@1.18.2~start: CWD: /opt/zigbee2mqtt
10 silly lifecycle zigbee2mqtt@1.18.2~start: Args: [ '-c', 'node index.js' ]
11 silly lifecycle zigbee2mqtt@1.18.2~start: Returned: code: 1  signal: null
12 info lifecycle zigbee2mqtt@1.18.2~start: Failed to exec start script
13 verbose stack Error: zigbee2mqtt@1.18.2 start: `node index.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:315:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:315:20)
13 verbose stack     at maybeClose (internal/child_process.js:1048:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
14 verbose pkgid zigbee2mqtt@1.18.2
15 verbose cwd /opt/zigbee2mqtt
16 verbose Linux 5.4.83-v7+
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start"
18 verbose node v14.16.1
19 verbose npm  v6.14.12
20 error code ELIFECYCLE
21 error errno 1
22 error zigbee2mqtt@1.18.2 start: `node index.js`
22 error Exit status 1
23 error Failed at the zigbee2mqtt@1.18.2 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

I also tried changing the last line: port: null for autodetection) to:  port: /dev/ttyUSB0
Not sure if this is correct, but that's the closest I find:
![image](https://user-images.githubusercontent.com/79019508/116264111-504cef00-a77a-11eb-9883-a3389594dd91.png)
But that leads to the same errors ...

Do you have a clue? (I don't)

**Comment by Twit123** [comment]
You forgot the base_topic, it is required.

and FYI permit_join is a required topic

Si

**Reply by Ursusprimus** [reply]
I implemented your suggestions (and after getting an indentation error I added one space in front of reject_unauthorized: true) after which things look better:
![image](https://user-images.githubusercontent.com/79019508/116293148-d840f200-a796-11eb-8705-8f732ee2e20b.png)
However, the lower part indicates other errors ...
And the command  "node --trace-warnings ..." gets this:
![image](https://user-images.githubusercontent.com/79019508/116293441-2e159a00-a797-11eb-9c63-100e15f1490d.png)

What could be the next step?

**Comment by Twit123** [comment]
I still think its your config. try with

server: 'mqtt://127.0.0.1:1883'

And the ' '  are important.

Si

**Comment by Ursusprimus** [comment]
Thank you! I got it working now
